source/common/stats/timespan_impl.cc

```diff
@@ -0,0 +1,47 @@
+#include "common/stats/timespan_impl.h"
+
+namespace Envoy {
+namespace Stats {
+
+HistogramCompletableTimespanImpl::HistogramCompletableTimespanImpl(Histogram& histogram,
+                                                                   TimeSource& time_source)
+    : time_source_(time_source), histogram_(histogram), start_(time_source.monotonicTime()) {
+  if (!histogram.active()) {
```

Comment:
What is the purpose of the `active()` method in general and why do we check it here? This constructor doesn't do anything AFAICT so I'm confused about the purpose.

Reply:
This was likely the result of suggestion I made earlier to differentiate null histograms for the purpose of avoiding the asserts below.

But another thing that occurred to me is just add Histogram::Unit::NullHistogram to the enum and put that in the switch.

Reply:
I followed @jmarantz's suggestion here and made it an enum value instead, let me know what you all think, can revert back.